common/concurrency/WorkerPool.h

```diff
@@ -5,6 +5,25 @@
 namespace sorbet {
 class WorkerPool {
 public:
+    class MultiplexCleanup {
+    public:
+        MultiplexCleanup() = default;
+
+        MultiplexCleanup(MultiplexCleanup &&) = default;
+        MultiplexCleanup &operator=(MultiplexCleanup &&) = default;
+
+        MultiplexCleanup(const MultiplexCleanup &) = delete;
+        MultiplexCleanup &operator=(const MultiplexCleanup &) = delete;
+
+        // It would be neat if we could say that this method can only be called once.
+        // One way to do that would be to make this a static method somewhere, and accept the
+        // receiver as a && value. But clang currently won't check use-after-move errors in Sorbet,
+        // so there's no point to factoring it like that IMO.
+        void cleanup(WorkerPool &workers) {
+            workers.consumeCounters();
+        }
```

Comment:
Note that nothing ensures this is actually called; you may consider adding a boolean flag that gets set to true in this function and `ENFORCE`'d to be true in the destructor.